src/lib/state.ts

```diff
@@ -578,6 +587,23 @@ export const dumpNode = (node: ZWaveNode, schemaVersion: number): NodeState => {
   return node14 as NodeStateSchema15;
 };
 
+export const dumpFoundNode = (
+  foundNode: ZWaveNode,
+  schemaVersion: number
+): FoundNodeState => {
+  const base: FoundNodeStateSchema19 = {
+    nodeId: foundNode.nodeId,
+    deviceClass: foundNode.deviceClass
+      ? dumpDeviceClass(foundNode.deviceClass)
+      : null,
+    status: foundNode.status,
+    commandClasses: Array.from(foundNode.getSupportedCCInstances(), (cc) =>
```

Comment:
this is safe right? From the code it seems like the CC map is initialized so worst case this will be empty?

Reply:
This should be safe.

FWIW using `getSupportedCCInstances` is an unnecessary complicated way to get to the CC information, since it creates an instance of each CC, but there's nothing better exposed atm.

When v10 is released, you can use `node.getCCs()` instead, which just returns the ID and version/supported/controlled/secure info.